[#6110] doc(authz): Add document for chain authorization plugin (#6115)
apache/gravitino: docs/security/authorization-pushdown.md

:::

### chain authorization plugin

Gravitino supports chaining multiple authorization plugins to secure one catalog.
The authorization plugin chain is defined in the `authorization.chain.plugins` property, with the plugin names separated by commas.
When a user performs an authorization operation on data within a catalog, the chained plugin will apply the authorization rules for every plugin defined in the chain.

In order to use the chained authorization plugin, you need to configure the following properties:

| Property Name                                             | Description                                                                               | Default Value | Required                    | Since Version    |
|-----------------------------------------------------------|-------------------------------------------------------------------------------------------|---------------|-----------------------------|------------------|
| `authorization-provider`                                  | Providers to use to implement authorization plugin such as `chain`                        | (none)        | No                          | 0.8.0-incubating |

> | `authorization.chain.plugins`                             | The comma-separated list of plugin names, just like `${plugin-name1},${plugin-name2},...` | (none)        | Yes if you use chain plugin | 0.8.0-incubating |

Reviewer: ```suggestion
| `authorization.chain.plugins`                             | The comma-separated list of plugin names, like `${plugin-name1},${plugin-name2},...` | (none)        | Yes if you use chain plugin | 0.8.0-incubating |
```

| `authorization.chain.${plugin-name}.ranger.service.type`  | The Ranger authorization plugin properties of the `${plugin-name}`                        | (none)        | Yes if you use chain plugin | 0.8.0-incubating |
| `authorization.chain.${plugin-name}.ranger.service.name`  | The Ranger authorization plugin properties of the `${plugin-name}`                        | (none)        | Yes if you use chain plugin | 0.8.0-incubating |
| `authorization.chain.${plugin-name}.ranger.username`      | The Ranger authorization plugin properties of the `${plugin-name}`                        | (none)        | Yes if you use chain plugin | 0.8.0-incubating |
| `authorization.chain.${plugin-name}.ranger.password`      | The Ranger authorization plugin properties of the `${plugin-name}`                        | (none)        | Yes if you use chain plugin | 0.8.0-incubating |

:::caution
The Gravitino chain authorization plugin only supports the Apache Ranger HadoopSQL Plugin and Apache Ranger HDFS Plugin.

> The properties of every chained plugin in the authorization should be configured with the `authorization.chain.${plugin-name}` as the prefix.

Reviewer: ```suggestion
The properties of every chained authorization plugin should use `authorization.chain.${plugin-name}` as the prefix.
```

:::

#### Example of using the chain authorization Plugin

Suppose you have an Apache Hive service in your datacenter and have created a `hiveRepo` in Apache Ranger to manage its permissions.

> The Apache Hive service will use HDFS to store its data. You have created a `hdfsRepo` in Apache Ranger to manage HDFS's permissions.

Reviewer: What does "You have created" mean?
The four (sub-)sentences there are all assumptions?

Author: Yes, they are assumptions.
